packages/sdk/src/protocols/message_reliability_monitor.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  IFilterSDK,
+  IProtoMessage,
+  ISubscriptionSDK,
+  PeerIdStr,
+  PubsubTopic
+} from "@waku/interfaces";
+import { messageHashStr } from "@waku/message-hash";
+import { WakuMessage } from "@waku/proto";
+import { Logger } from "@waku/utils";
+
+const log = new Logger("sdk:message_reliability_monitor");
+
+const DEFAULT_MAX_MISSED_MESSAGES_THRESHOLD = 3;
+
+export class MessageReliabilityManager {
+  public constructor(private filter: IFilterSDK) {
+    this.filter.activeSubscriptions.forEach((subscription) => {
+      new MessageReliabilityMonitor(this.filter, subscription);
```

Comment:
also let's not loose references in case we need to utilize these objects or use later